components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -256,37 +283,53 @@ def compress(
     clp_io_config = ClpIoConfig.parse_raw(clp_io_config_json)
     paths_to_compress = PathsToCompress.parse_raw(paths_to_compress_json)
 
-    start_time = datetime.datetime.now()
-    logger.info(f"[job_id={job_id} task_id={task_id}] COMPRESSION STARTED.")
-    compression_successful, worker_output = run_clp(
-        clp_io_config,
-        pathlib.Path(clp_home_str),
-        pathlib.Path(data_dir_str),
-        pathlib.Path(archive_output_dir_str),
-        pathlib.Path(logs_dir_str),
-        job_id,
-        task_id,
-        tag_ids,
-        paths_to_compress,
-        clp_metadata_db_connection_config,
-    )
-    duration = (datetime.datetime.now() - start_time).total_seconds()
-    logger.info(f"[job_id={job_id} task_id={task_id}] COMPRESSION COMPLETED.")
+    sql_adapter = SQL_Adapter(Database.parse_obj(clp_metadata_db_connection_config))
 
-    if compression_successful:
-        return CompressionTaskSuccessResult(
-            task_id=task_id,
-            status=CompressionTaskStatus.SUCCEEDED,
-            start_time=start_time,
-            duration=duration,
-            total_uncompressed_size=worker_output["total_uncompressed_size"],
-            total_compressed_size=worker_output["total_compressed_size"],
-        ).dict()
-    else:
-        return CompressionTaskFailureResult(
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
```

Comment:
I still think we should try to keep our MySQL connections short, even if frequent; so I would prefer we open a connection only just before we perform a write (currently, this opens a connection before we start compression, which itself could take a long time depending on the dataset and config).

MySQL's [blog](https://dev.mysql.com/blog-archive/mysql-connection-handling-and-scaling/) says that it's capable of handling a lot of short connections, but its default concurrent connection limit is only [151](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_connections). We should probably benchmark this for ourselves at some point (a long time ago, @haiqi96 had done some scalability benchmarking that showed MySQL struggled with 20 concurrent connections performing inserts), but for now, I think following their advice is the safer option.

Reply:
Yeah, my original plan was to use short connection. But I notice in compression scheduler, we maintain a long connection.

Reply:
In the scheduler's case, it's only maintaining one connection that it's using for polling (among other things), right? In theory we could make it use shorter connections, but there I'm not sure it will make much difference (we should still measure at some point though).

Reply:
Got it. For the compression task, we read the output from the process and update the database. Do you think we should open a connection each time when we get a new line?

Reply:
I think we could try opening a connection each time we need to update the archive's stats + tags (which would only be every time we finish an archive) and then once at the end of the task.